core/src/main/java/ai/timefold/solver/core/impl/solver/DefaultSolverJob.java

```diff
@@ -306,4 +311,24 @@ public void phaseEnded(AbstractPhaseScope<Solution_> phaseScope) {
             }
         }
     }
+
+    /**
+     * A listener that is triggered once when the solver starts the solving process.
+     */
+    private final class StartSolverJobPhaseLifecycleListener extends PhaseLifecycleListenerAdapter<Solution_> {
```

Comment:
`SolverLifecycleListenerAdapter` can listen to `solvingStarted` events. This is what you should listen to, instead of reading the phases.